src/Sulu/Component/Content/Document/Subscriber/StructureRemoveSubscriber.php

```diff
@@ -57,19 +58,19 @@ public function handleRemove(RemoveEvent $event)
 
     public function removeDocument($document)
     {
-        // TODO: This is not a good indicator. There should be a RoutableBehavior here.
-        if (!$document instanceof StructureBehavior) {
-            return;
-        }
-
         if ($document instanceof ChildrenBehavior) {
             foreach ($document->getChildren() as $child) {
                 $this->removeDocument($child);
             }
         }
 
-        $this->removeReferences($document);
-        $this->recursivelyRemoveRoutes($document);
+        if ($document instanceof StructureBehavior) {
+            $this->removeReferences($document);
+        }
+
+        if ($document instanceof RoutableBehavior) {
```

Comment:
Would be ok for me, if we also do that before the 1.2 release (otherwise it's not done anyway), and we would solve it for now in a different way than introducing another behavior (doesn't make sense to me to introduce a new interface here, if we know that we'll remove it later).